ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -100,13 +99,8 @@ def process_result(
         """
         result_buf = StringIO()
 
-        # policy breaks and matches are modified in the functions leak_dictionary_by_ignore_sha and censor_content.
```

Comment:
`leak_dictionary_by_ignore_sha()` still modifies scan.policy_breaks, so I think we should continue creating a copy of result.

Reply:
Hum if I understand well it mostly modifies the order of policy breaks and the order of the matches in the policy break ? It seems to be called several time redundantly including as part of the init of [SecretScanCollection](https://github.com/GitGuardian/ggshield/blob/5eeff39a8ab083f78db1a51720cbd22d4ec36c28/ggshield/verticals/secret/secret_scan_collection.py#L161-L161).

For coherence across execution, I think I would prefer always enforcing this order rather than copying the result but if you disagree I will trust your guts more than mine ! :smile:

Reply:
Oh, I missed that it was called some other places. This prompted me to make the experiment or removing all sorting from that method and... it seems to be fine! No tests broke.

Going to file a PR to just remove sorting, this way `leak_dictionary_by_ignore_sha()` no longer has any side-effect. Let's keep your code the way it is.